utils/log_validator.py

```diff
@@ -37,10 +37,13 @@ class LogValidator(object):
     """
 
     def __init__(self, remote_filename, **kwargs):
-        self._remote_file_tail = SSHTail(remote_filename)
-        self.skip_patterns = kwargs['skip_patterns'] if 'skip_patterns' in kwargs else []
-        self.failure_patterns = kwargs['failure_patterns'] if 'failure_patterns' in kwargs else []
-        self.matched_patterns = kwargs['matched_patterns'] if 'matched_patterns' in kwargs else []
+        self.skip_patterns = kwargs.pop('skip_patterns') if 'skip_patterns' in kwargs else []
```

Comment:
Suggest you use kwargs built-in get() method, which takes a default value as a second parameter.

```python
self.var = kwargs.get('var', [])
```